graphics/ImageMagick7/Portfile

```diff
@@ -1,193 +1,77 @@
-                    --without-gcc-arch \
-                    --without-perl \
-                    --without-fpx \
-                    --without-wmf \
-                    --without-gvc \
-                    --without-rsvg \
-                    --without-lqr \
-                    --without-pango \
-                    --without-x \
-                    --without-zstd \
-                    --with-gs-font-dir=${prefix}/share/fonts/urw-fonts \
-                    --disable-openmp
-
-if {${os.platform} eq "darwin" && ${os.major} < 11} {
-    configure.args-append   --disable-opencl
-} else {
-    configure.args-append   --disable-opencl
-    # On case-insensitive filesystems, ImageMagick finds cryptlib's libcl and
-    # tries to use it as if it were Apple's OpenCL, which fails; see #23354.
-    if {[file exists ${prefix}/lib/libCL.dylib]} {
-        conflicts_build         cryptlib
-    }
-}
-
-test.run            yes
-test.target         check
-test.env            DYLD_LIBRARY_PATH=${worksrcpath}/magick/.libs
-
-# ImageMagick uses .la files at runtime to find its coder modules.
-destroot.delete_la_files    no
-
-variant graphviz description {Support Graphviz} {
-    depends_lib-append      path:bin/dot:graphviz
-    configure.args-replace  --without-gvc --with-gvc
-}
+PortGroup           meson 1.0
+PortGroup           gobject_introspection 1.0
+PortGroup           gitlab 1.0
 
-variant lqr description {Support Liquid Rescale (experimental)} {
-    depends_lib-append      port:liblqr
-    configure.args-replace  --without-lqr --with-lqr
-}
+gitlab.instance     https://gitlab.gnome.org
+gitlab.setup        GNOME gjs 1.78.4
+revision            0
 
-variant pango description {Support Pango} {
-    depends_lib-append      path:lib/pkgconfig/pango.pc:pango
-    configure.args-replace  --without-pango --with-pango
+checksums           rmd160  dad53f8fad8ea45ec1077f31c1d4449bbbae68fe \
+                    sha256  dc635cb481047dbdee3395f3ff3ac121e8449d5fce71682ad0a21df503237a3b \
+                    size    691601
+ 
+conflicts           gjs-devel
+set my_name         gjs
+
+categories          gnome
+license             LGPL-2.1+
+maintainers         {devans @dbevans} {mascguy @mascguy} openmaintainer
+
+description         GNOME JavaScript/Spidermonkey bindings
+long_description    ${description}
+homepage            https://gitlab.gnome.org/GNOME/gjs/-/blob/master/doc/README.md \
+                    https://gjs-docs.gnome.org/gjs
+
+set py_ver          3.12
+set py_ver_nodot    [string map {. {}} ${py_ver}]
+
+depends_build-append \
+                    port:pkgconfig \
+                    port:python${py_ver_nodot}
+
+depends_lib-append \
+                    path:lib/pkgconfig/cairo.pc:cairo \
+                    port:dbus \
+                    path:lib/pkgconfig/glib-2.0.pc:glib2 \
+                    path:lib/pkgconfig/gtk+-3.0.pc:gtk3 \
+                    port:libffi \
+                    port:mozjs115 \
```

Comment:
Could we please avoid depending by default on a port which is broken for many systems: https://ports.macports.org/port/mozjs115/details
That could be done via a conditional variant.
(Why does it even need javascript engine?)